site/zenodo_rdm/legacy/deserializers/schemas.py

```diff
@@ -238,11 +248,203 @@ def _additional_descriptions(self, data):
+        method = data.get("method")
+        if method:
+            rdm_additional_descriptions.append(
+                {"description": method, "type": {"id": "methods"}}
+            )
 
         return rdm_additional_descriptions
 
+    def _locations(self, data):
+        """Transform locations of a legacy record."""
+        if not data:
+            return None
+
+        features = []
+        for legacy_location in data:
+            lat = legacy_location.get("lat")
+            lon = legacy_location.get("lon")
+            place = legacy_location["place"]
+            description = legacy_location.get("description")
+
+            feature = {"place": place}
+
+            if lat and lon:
+                geometry = {"type": "Point", "coordinates": [lon, lat]}
+                feature.update({"geometry": geometry})
+
+            if description:
+                feature.update({"description": description})
+
+            features.append(feature)
+
+        return {"features": features}
+
+    def _subjects(self, data):
+        """Transform subjects of a legacy record.
+
+        RDM subjects translate to either legacy keywords or subjects.
+        """
+
+        def _subjects_from_keywords(keywords):
+            """Legacy keywords are free text strings.
+
+            They map to custom subjects.
+            """
+            return [{"subject": kw} for kw in keywords]
+
+        def _subjects_from_subjects(data):
+            """Maps RDM subjects to legacy subjects.
+
+            Legacy subjects are custom vocabularies.
+            """
+            # TODO we still did not define a strategy to map legacy subjects to rdm.
+            return None
+
+        keywords = data.get("keywords", [])
+        subjects = data.get("subjects", [])
+
+        if not (keywords or subjects):
+            return None
+
+        rdm_subjects = []
+
+        kws = _subjects_from_keywords(keywords)
+        if kws:
+            rdm_subjects.extend(kws)
+
+        sbjs = _subjects_from_subjects(subjects)
+        if sbjs:
+            rdm_subjects.extend(sbjs)
+
+        return rdm_subjects
+
+    def _dates(self, data):
+        """Transform dates of a legacy record."""
+        if not data:
+            return None
+
+        rdm_dates = []
+
+        for legacy_date in data:
+            start_date = legacy_date.get("start")
+            end_date = legacy_date.get("end")
+            description = legacy_date.get("description")
+            rdm_date = None
+            if start_date and end_date:
+                if start_date == end_date:
+                    rdm_date = start_date
+                else:
+                    rdm_date = f"{start_date}/{end_date}"
+            elif start_date:
+                rdm_date = start_date
+
+            elif end_date:
+                rdm_date = end_date
+            # No start_date nor end_date is not allowed on legacy Zenodo
```

Comment:
Maybe I missunderstood your comment but I think legacy requires at least one date to be set:

```python
if not start and not end:
    raise ValidationError()
```

I added the comment in the code to explain that no `else` clause was added because legacy does not allow that scenario (no start and no end date). In fact, in this case the legacy api should also return an error.

Open intervals are another thing. Our problem with open intervals is  that we can't serialize an RDM date to a legacy date when the interval is open (and for this we need EDTF Level 1, as you said.)